src/cpp/src/text2image/stable_diffusion_3_pipeline.hpp

```diff
@@ -0,0 +1,576 @@
+                                           transformer_config.joint_attention_dim};
+
+        std::vector<float> t5_prompt_embed(
+            t5_prompt_embed_shape[0] * t5_prompt_embed_shape[1] * t5_prompt_embed_shape[2],
+            0.0f);
+
+        // padding for clip_prompt_embeds
+        ov::Shape pad_embeds_shape = {clip_prompt_embeds_shape[0],
+                                      clip_prompt_embeds_shape[1],
+                                      t5_prompt_embed_shape[2]};
+
+        std::vector<float> pad_embeds(pad_embeds_shape[0] * pad_embeds_shape[1] * pad_embeds_shape[2], 0.0f);
+        padding_right(clip_prompt_embeds_data, pad_embeds.data(), clip_prompt_embeds_shape, pad_embeds_shape);
+
+        // prompt_embeds = torch.cat([pad_embeds, t5_prompt_embed], dim=-2)
+        ov::Shape prompt_embeds_shape = {pad_embeds_shape[0],
+                                         pad_embeds_shape[1] + t5_prompt_embed_shape[1],
+                                         pad_embeds_shape[2]};
+        ov::Tensor prompt_embeds(ov::element::f32, prompt_embeds_shape);
+        float* prompt_embeds_data = prompt_embeds.data<float>();
+        concat_3d_by_cols(pad_embeds.data(),
+                          t5_prompt_embed.data(),
+                          prompt_embeds_data,
+                          pad_embeds_shape,
+                          t5_prompt_embed_shape);
+
+        // pooled_prompt_embeds = torch.cat([pooled_prompt_embed, pooled_prompt_2_embed], dim=-1)
+        ov::Shape p_pr_emb_shape = pooled_prompt_embed.get_shape();
+        ov::Shape p_pr_emb_2_shape = pooled_prompt_2_embed.get_shape();
+
+        const float* pooled_prompt_embed_data = pooled_prompt_embed.data<float>();
+        const float* pooled_prompt_2_embed_data = pooled_prompt_2_embed.data<float>();
+
+        ov::Shape pooled_prompt_embeds_shape = {p_pr_emb_shape[0], p_pr_emb_shape[1] + p_pr_emb_2_shape[1]};
+        ov::Tensor pooled_prompt_embeds(ov::element::f32, pooled_prompt_embeds_shape);
+        float* pooled_prompt_embeds_data = pooled_prompt_embeds.data<float>();
+
+        concat_2d_by_rows(pooled_prompt_embed_data,
+                          pooled_prompt_2_embed_data,
+                          pooled_prompt_embeds_data,
+                          p_pr_emb_shape,
+                          p_pr_emb_2_shape);
+        // From steps above we'll use prompt_embeds and pooled_prompt_embeds tensors
+
+        if (do_classifier_free_guidance(generation_config.guidance_scale)) {
+            // 2. Encode negative prompt:
+            std::string negative_prompt_1_str =
+                !generation_config.negative_prompt.empty() ? generation_config.negative_prompt : "";
+            std::string negative_prompt_2_str = !generation_config.negative_prompt_2.empty()
+                                                    ? generation_config.negative_prompt_2
+                                                    : negative_prompt_1_str;
```

Comment:
looks like we need to have type of `prompt_2` and `prompt_3` as `std::optional<std::string>` instead of `std::string` to distinguish empty negative prompt from un-initialized negative prompt.

Let's do it here https://github.com/openvinotoolkit/openvino.genai/blob/master/src/cpp/include/openvino/genai/text2image/pipeline.hpp#L75